### ISSUE TITLE
nodetool: permission issues when running nodetool

### ISSUE THREAD

**Issue by eliransin** [scylladb/scylla-tools-java#147]
This has been discovered by:
https://jenkins.scylladb.com/job/scylla-master/job/artifacts/job/artifacts-centos7/34/
and a bunch of other artifact test on other distributions.
At some point, when running nodetool, cassandra-env.sh runs, this results in a failure on permission denied.
A recent commit : afd40c7 have added a creation of logs directory for the garbage collector and it 
tries to create it in $CASSANDRA_HOME those are the relevant lines:
https://github.com/scylladb/scylla-tools-java/blob/d3e5bbe208e1fd3dea7f4bdf82489c14b2f3e210/conf/cassandra-env.sh#L89-L108

In this particular failure, there is an attempt to create: /opt/scylladb/share/cassandra/logs which fails
on permission denied.

```
17:46:27  < t:2020-03-29 14:46:24,571 f:sct_events.py   l:791  c:sdcm.sct_events      p:INFO  > mkdir: cannot create directory ‘/opt/scylladb/share/cassandra/logs’: Permission denied
```
AFAICT this is a new behavior (the dir creation) but it might be necessary. So the options are:
1. (@syuu1228) the permissions on   `/opt/scylladb/share/cassandra` are not right.
2. (@elcallio) maybe there is something else that needs to be merged with this
3. There is a restriction on the user that should run nodetool (sudo / scylla user) - I couldn't find this in the docs.

/cc @roydahan

**Comment by penberg** [comment]
Likely fixed by the patch `[PATCH scylla-tools v1 1/3] dist/common: Add Java version detection to cassandra.in.sh` on our mailing list.

**Comment by penberg** [comment]
Btw, the bug was introduced in 3884643859e387040c1eeaf739f850355bc23aba, not sure why bring-your-own dtest did not catch it.

**Comment by amoskong** [comment]
Hi @penberg I can still see the permission error with latest fixed scylla-tools-java.

#### DB log:
https://cloudius-jenkins-test.s3.amazonaws.com/13892b84-bf1d-47d7-9110-d82dea9bc5b5/20200409_171438/db-cluster-13892b84.zip 

test id: 13892b84-bf1d-47d7-9110-d82dea9bc5b5
test job: https://jenkins.scylladb.com/job/scylla-staging/job/amos/job/longevity-large-collections-12h-clone/14/
sct log:  https://cloudius-jenkins-test.s3.amazonaws.com/13892b84-bf1d-47d7-9110-d82dea9bc5b5/20200409_171438/sct-runner-13892b84.zip

Scylla-tools-java commit: https://github.com/scylladb/scylla-tools-java/commit/2f73d3ae377

```
`rpm -qa |grep scylla` on longevity-large-collections-12h-lar-db-node-13892b84-1 [34.250.63.9 | 10.0.139.95
- scylla-python3-3.7.7-0.20200405.df48b5ec9d5.x86_64
- scylla-kernel-conf-666.development-0.20200405.df48b5ec9d5.x86_64
- scylla-conf-666.development-0.20200405.df48b5ec9d5.x86_64
- scylla-jmx-666.development-20200406.773a82d5396.noarch
- scylla-debuginfo-666.development-0.20200405.df48b5ec9d5.x86_64
- scylla-tools-666.development-20200406.2f73d3ae377.noarch
- scylla-machine-image-666.development-20200406.b06535de7a4.noarch
- scylla-server-666.development-0.20200405.df48b5ec9d5.x86_64
- scylla-tools-core-666.development-20200406.2f73d3ae377.noarch
- scylla-666.development-0.20200405.df48b5ec9d5.x86_64
```

```
04-09 04:19:52 [centos@10.0.139.95]: Running command "nodetool  status "...
04-09 04:19:52 [centos@10.0.139.95]: mkdir: cannot create directory ‘/opt/scylladb/share/cassandra/logs’: Permission denied
04-09 04:19:54 [centos@10.0.139.95]: Datacenter: eu-west
04-09 04:19:54 [centos@10.0.139.95]: ===================
04-09 04:19:54 [centos@10.0.139.95]: Status=Up/Down
04-09 04:19:54 [centos@10.0.139.95]: |/ State=Normal/Leaving/Joining/Moving
04-09 04:19:54 [centos@10.0.139.95]: --  Address      Load       Tokens       Owns    Host ID                               Rack
04-09 04:19:54 [centos@10.0.139.95]: UN  10.0.139.95  284.97 KB  256          ?       9354fe86-4c5f-473f-a07c-1a7f49d22ca2  1a
04-09 04:19:54 [centos@10.0.139.95]:
04-09 04:19:54 [centos@10.0.139.95]: Note: Non-system keyspaces don't have the same replication settings, effective ownership information is meaningless
04-09 04:19:54 [centos@10.0.139.95]: STDERR: mkdir: cannot create directory ‘/opt/scylladb/share/cassandra/logs’: Permission denied
04-09 04:19:54 [centos@10.0.139.95]: Command "nodetool  status " finished with status 0
```


```
Decommission nemesis:
Node longevity-large-collections-12h-lar-db-node-13892b84-2 [34.250.153.111 | 10.0.246.170] (seed: False): Command 'nodetool  cleanup large_collection_test' error:          Encountered a bad command exit code!

| Command: 'nodetool  cleanup large_collection_test'

| Exit code: -1

| Stdout:

| Stderr:

mkdir: cannot create directory ‘/opt/scylladb/share/cassandra/logs’: Permission denied
WARN  15:41:11,656 Small cdc volume detected at /cdc_raw; setting cdc_total_space_in_mb to 1918.  You can override this in cassandra.yaml
WARN  15:41:11,662 Only 9.122GiB free across all data volumes. Consider adding more capacity to your cluster or removing obsolete snapshots
```

**Comment by penberg** [comment]
Oh, interesting. The fixes for this issues were related to JVM detection (`nodetool` didn't run at all), but looks like that "permission denied" issue was another, possibly unrelated issue. Let me reopen this ticket as suggested by @fruch.

**Comment by penberg** [comment]
@elcallio Perhaps the Java 11 support merge broke something? Are we even supposed to write logs to `/opt/scylladb/share/cassandra/logs`?

**Comment by eliransin** [comment]
@fruch is the test uses nodetool as it should (as root or as scylla:scylla user?)

**Comment by fruch** [comment]
@eliransin No it's running as `centos` user, as it was since forever :disappointed:

**Comment by eliransin** [comment]
@slivne / @tzach  it is my understanding that all of the scylla-tools-java binaries should be either run with root permissions or under scylla:scylla user. is this correct? 
If not, we should fix it as it is brought up too many times (see #118) . It this is what suppose to be, I suggest to add this limitation to the documentation and in parallel maybe edit cassandra-env.sh to warn if it detects it is run without the proper permissions.

**Comment by eliransin** [comment]
@slivne corrected me, there shouldn't be any permissions issue with nodetool. There are some other tools that might have to be run with certain permissions (i.e sstabledump) but that is because it needs an access to the scylla data folder which is owned by user scylla:scylla. Nodetool doesn't need to access any directory ( except for logging purposes which is what we see here) so it is a bug
in nodetool and not in the documentation.

**Comment by eliransin** [comment]
OK, it is obvious that this logs folder which is always created in: $CASSANDRA_HOME/logs,
which translates in our case to: 
/opt/scylladb/share/cassandra/logs
I am pretty sure that we should do several things in the context of this issue:
1. To create this logs folder in advance (at install time and not at runtime)
2. In order to do this the logs directory write permissions should match to nodetool execute permissions. (or the logging can't happen)
3. At runtime we can add a check if this folder exists, if not then attempt to create it, if we can't, issue a warning and skip the gc logging. The only thing we will lose from it is the java gc logs not being created.
@penberg @elcallio any thoughts about that?

**Comment by amoskong** [comment]
We can see this issue in 4.1.rc0 testing, (4.1.rc0-0.20200511.0760107b9fa)
- https://jenkins.scylladb.com/view/scylla-4.1/job/scylla-4.1/job/artifacts/job/artifacts-ami/23
- https://jenkins.scylladb.com/job/scylla-4.1/job/longevity/job/longevity-large-partition-8h/2/

**Comment by slivne** [comment]
fixed by https://github.com/scylladb/scylla-tools-java/commit/80e8dab844262139bc0bd2b0ed8ce12cf43a26de

**Comment by eliransin** [comment]
@slivne do you want this backported?

**Comment by slivne** [comment]
already done 6ce3dd19252545c8ea74522a45f1e87225d1d53e :)
